### PR TITLE
[12.0][ADD] hr_contract_currency: non-default currency for contract

### PR DESCRIPTION
**Pull request by alexey-pelykh**
image

### CODE REVIEW

File: hr_contract_currency/models/hr_contract.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    currency_id = fields.Many2one(
```
Comment (alexey-pelykh) [inline hr_contract_currency/models/hr_contract.py:10]: @pedrobaeza is this correct way of converting a related field to be non-related?

Comment (alexey-pelykh) [reply hr_contract_currency/models/hr_contract.py:10]: another question: for some reason, changes on this field are not "tracked" in the bottom of the screen:
image
a hint would be much appreciated

Comment (pedrobaeza) [reply hr_contract_currency/models/hr_contract.py:10]: Yes, although changing fields this way sometimes can be a problem with other modules. Let's hope it isn't in this case. For having that changes, you have to add `track_visibility` argument

Comment (alexey-pelykh) [reply hr_contract_currency/models/hr_contract.py:10]: Awesome, thanks!

**Comment by alexey-pelykh** [comment]
This one will be simple and separate from hr_payroll_currency

File: hr_contract_currency/views/hr_contract.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            </xpath>
+        </field>
+    </record>
+
+    <record id="hr_contract_view_tree" model="ir.ui.view">
+        <field name="name">hr.contract.view.tree.inherit.currency</field>
+        <field name="model">hr.contract</field>
+        <field name="inherit_id" ref="hr_contract.hr_contract_view_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='wage']" position="after">
```
Comment (nikul-serpentcs) [inline hr_contract_currency/views/hr_contract.xml]: @alexey-pelykh use field directly

**Review by nikul-serpentcs — APPROVED** [review]
Minor change Otherwise LGTM

File: hr_contract_currency/views/hr_contract.xml (hunk 2 above)
Comment (alexey-pelykh) [reply hr_contract_currency/views/hr_contract.xml]: Thanks, updated!

**Comment by alexey-pelykh** [comment]
@feketemihai @mreficent I'd like to request your reviews as well, if you don't mind

**Comment by AaronHForgeFlow** [comment]
Is the currency field passed to payslips? or this is just info?

**Comment by alexey-pelykh** [comment]
@aheficent the field is there initially:
https://github.com/odoo/odoo/blob/9dbb7d199cfe085ea5e3a4980de2bd766953ab93/addons/hr_contract/models/hr_contract.py#L79
but it's set as being mirror of default company currency.

The `hr_payroll` does not include currency or conversion at all, to answer your question directly - that's something different to address:
https://github.com/odoo/odoo/blob/9c3ec026c541901a9fe75db9bca5f9850276cb41/addons/hr_payroll/data/hr_payroll_data.xml#L46

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Thanks for the explanation. LGTM.

**Pull request merged**